src/overlays/actors/ovl_Boss_07/z_boss_07.c

```diff
@@ -242,500 +636,7211 @@ ActorProfile Boss_07_Profile = {
+            Math_ApproachF(&this->subCamAt.y, this->actor.world.pos.y, 0.05f, 10.0f);
+            Math_ApproachF(&this->subCamAt.z, this->actor.world.pos.z, 0.05f, 10.0f);
+            break;
+
+        default:
+            break;
+    }
+    if (this->subCamId != SUB_CAM_ID_DONE) {
+        Vec3f subCamAt;
+
+        ShrinkWindow_Letterbox_SetSizeTarget(27);
+        subCamAt.x = this->subCamAt.x;
+        subCamAt.y = this->subCamAt.y + subCamAtYOffset;
+        subCamAt.z = this->subCamAt.z;
+        Play_SetCameraAtEye(play, this->subCamId, &subCamAt, &this->subCamEye);
+    }
+}
+
+void Boss07_Mask_SetupDeathCutscene(Boss07* this, PlayState* play) {
+    this->actionFunc = Boss07_Mask_DeathCutscene;
+    Animation_MorphToLoop(&this->skelAnime, &gMajorasMaskJerkingAnim, -10.0f);
+    this->actor.world.rot.y = this->actor.shape.rot.y =
+        Math_Atan2F_XY(-this->actor.world.pos.z, -this->actor.world.pos.x) * (0x8000 / M_PIf);
+    this->cutsceneState = MAJORAS_MASK_DEATH_CS_STATE_STARTED;
+    this->cutsceneTimer = 0;
+    this->actor.flags &= ~ACTOR_FLAG_ATTENTION_ENABLED;
+    this->damagedTimer = 20;
+}
+
+void Boss07_Mask_DeathCutscene(Boss07* this, PlayState* play) {
+    Player* player = GET_PLAYER(play);
+    Vec3f subCamAtTarget;
+    Vec3f subCamEye;
+
+    this->cutsceneTimer++;
+    this->damagedTimer = 20;
+    this->maskEyeTexIndex = MAJORAS_MASK_EYE_DULL;
+    SkelAnime_Update(&this->skelAnime);
+    Boss07_SmoothStop(this, 0.5f);
+
+    Math_ApproachF(&this->actor.world.pos.x, 0.0f, 0.05f, 5.0f);
+    Math_ApproachF(&this->actor.world.pos.z, 0.0f, 0.05f, 5.0f);
+    Math_ApproachF(&this->actor.world.pos.y, 130.0f, 0.05f, 3.0f);
+
+    Matrix_Translate(this->actor.world.pos.x, this->actor.world.pos.y, this->actor.world.pos.z, MTXMODE_NEW);
+    Matrix_RotateYS(this->actor.shape.rot.y, MTXMODE_APPLY);
+
+    switch (this->cutsceneState) {
+        case MAJORAS_MASK_DEATH_CS_STATE_STARTED:
+            if (CutsceneManager_GetCurrentCsId() == CS_ID_NONE) {
+                Cutscene_StartManual(play, &play->csCtx);
+                Player_SetCsActionWithHaltedActors(play, &this->actor, PLAYER_CSACTION_1);
+                this->subCamId = Play_CreateSubCamera(play);
+                Play_ChangeCameraStatus(play, CAM_ID_MAIN, CAM_STATUS_WAIT);
+                Play_ChangeCameraStatus(play, this->subCamId, CAM_STATUS_ACTIVE);
+                this->cutsceneTimer = 0;
+                this->cutsceneState = MAJORAS_MASK_DEATH_CS_STATE_PLAYING;
+                this->subCamAtNext.z = 0.0f;
+                this->fireTimer = 120;
+                Play_EnableMotionBlur(150);
+
+                this->subCamEyeNext.x = 0.0f;
+                this->subCamEyeNext.y = -30.0f;
+                this->subCamEyeNext.z = 270.0f;
+
+                this->subCamAtNext.x = 0.0f;
+                this->subCamAtNext.y = -30.0f;
+
+                Matrix_MultVec3f(&this->subCa
```

Comment:
Should this be considered a bug?

Reply:
oops didn't notice it was messed up

I mean here
https://github.com/zeldaret/mm/blob/9d3c7d51d2b1bda066d62b27c5e7d5890b8ad56f/src/overlays/actors/ovl_Boss_07/z_boss_07.c#L6866

Reply:
Before doing some research to respond to this, I was convinced that this was too difficult to tell. After all, who's to say that it was *intentionally* broken to disable this behavior? I was going to argue it would be almost impossible to tell without asking the original developers...

...until I tried it in MM3D and found they patched it lol. It's definitely a bug, so let me change that.

https://github.com/user-attachments/assets/750481ed-64fa-4b22-9d5a-83cd1687db73